crypto/x509/v3_san.c

```diff
@@ -183,9 +186,18 @@ STACK_OF(CONF_VALUE) *i2v_GENERAL_NAME(X509V3_EXT_METHOD *method,
         break;
 
     case GEN_DIRNAME:
-        if (X509_NAME_oneline(gen->d.dirn, oline, sizeof(oline)) == NULL
-                || !X509V3_add_value("DirName", oline, &ret))
+        bio = BIO_new(BIO_s_mem());
+        BIO_puts(bio, "/");
+        r = X509_NAME_print_ex(bio, gen->d.dirn, 0,
+                               (ASN1_STRFLGS_RFC2253 | ASN1_STRFLGS_ESC_QUOTE |
+                                XN_FLAG_SEP_SLASH_PLUS | XN_FLAG_FN_SN)
+                               & ~ASN1_STRFLGS_ESC_MSB);
+        BIO_get_mem_data(bio, &data);
```

Comment:
I (think) the stack allocated oline buffer is sized to ensure that it can hold a maximally sized DirName buffer (64 bytes I believe?).  The use of BIO_s_mem here, while sensible, isn't going to return enough data in the membuf obtained by BIO_get_mem_data to ensure that you can hold the data added by X509V3_add_data below.

I think thats whats causing the CI failures below.  You should be able to fix it by replacing the call to BIO_new(BIO_s_mem()) with a call to BIO_new_mem_buf(oline, sizeof(oline))

Reply:
Thanks for the tip. I'll need to firstly find out why all tests are passing on my M4, and then I'll submit the change.

Reply:
`BIO_new_mem_buf()` won't help in this case because it creates RDONLY bio from the `oline` which at this point empty. What puzzeles me is why `BIO_s_mem` isn't going to return enough data in the membuf obtained by `BIO_get_mem_data`.

Reply:
It seems that BIO_new(BIO_s_mem()) with string valid string termination '\0'  fix the problem.